components/core/src/clp_s/search/ast/ColumnDescriptor.hpp

```diff
@@ -280,6 +282,12 @@ class ColumnDescriptor : public Literal {
         m_namespace = descriptor_namespace;
     }
 
+    auto get_subtree_type() const -> std::optional<std::string> const& { return m_subtree_type; }
```

Comment:
Can we get rid of `get_node_type_for_subtree_type` and have a method `get_subtree_node_type` instead?